db/tasks.go

```diff
@@ -176,8 +176,7 @@ func (db *DB) GetTask(id uuid.UUID) (*Task, error) {
 	return r[0], nil
 }
 
-func (db *DB) CreateBackupTask(owner string, job *Job) (*Task, error) {
-	id := uuid.NewRandom()
+func (db *DB) CreateBackupTask(owner string, job *Job, id uuid.UUID) (*Task, error) {
```

Comment:
The `*Task* that this function previously returned should have had the generated UUID in it, so I don't think you need to forcibly impose a UUID from outside on the creation.  The proposed change also makes it so that every caller of `CreateBackupTask()` needs to be cognizant not to reuse UUIDs accidentally, which makes the overall system a little more brittle.